Add documentation

nchristensen opened the pull request:
Pull request to stage NekROM documentation. This uses sphinx-fortran and needs the bug fix at https://github.com/numpy/numpy/pull/28396.

nchristensen commented:
Instructions to build documentation:
1. Set up a new conda environment and activate it or use an existing conda environment if you don't mind using a development version of numpy
2. Clone this branch of numpy: https://github.com/nchristensen/numpy/tree/common-block-division
3. cd into the numpy directory and install it into the conda environment using these instructions: https://numpy.org/doc/stable//building/index.html#building-from-source-to-use-numpy
4. pip install sphinx and pip install sphinx-fortran
5. Clone this branch of NekROM: https://github.com/nchristensen/NekROM/tree/documentation
6. cd into NekROM/doc
7. run make html
8. To view the documentation, cd into NekROM/doc/build and open index.html in your web browser
9. Refactor source code following this style to make data show up in the documentation.

nchristensen commented:
@kent0 Should we just merge this PR as is as baseline documentation? It will be easier for other people to contribute to the docs once this is merged.

kent0 commented:
Yes, I think it's useful to include this as a baseline. Before merging, can you explain the changes to the `pod_proj` description?

nchristensen commented:
@kent0 I wasn't able to make Sphinx correctly associate comments on multiple lines with the `msg` parameter; it only does so for the first line and places the rest of the lines in the general subroutine documentation. As a workaround, I moved up the word "functions" into the first line so the displayed documentation for that parameter has a complete sentence and moved the description of the possible parameter values to the subroutine documentation. I assume we want to keep the line lengths at 80 characters, so I changed the wording a bit to make the sentence fit on one line.

kent0 commented:
I see, changing the wording is completely fine, but the rewording seems to be less clear than the original so how about "specifier for different ..." or something similar?

nchristensen commented:
Good idea. Made that change.
